lib/instrumentation/amqplib/utils.js

```diff
@@ -131,11 +112,33 @@ function setCallback(shim, promiseMode) {
   return promiseMode ? null : shim.LAST
 }
 
+/**
+ * Parses the connection args to return host/port
+ *
+ * @param {string|object} connArgs connection arguments
+ * @returns {object} {host, port }
+ */
+function parseConnectionArgs({ shim, connArgs }) {
+  const params = {}
+  if (shim.isString(connArgs)) {
+    connArgs = url.parse(connArgs)
```

Comment:
We have an outstanding issue to replace all instances of `url.parse`. Are you sure you want to add another instance here? Some string splitting or regular expression matching might be easier and faster.

Reply:
good point, i can fix these